### PR TITLE
update test-crypto.lua for lua-openssl v0.7.4

### PR DESCRIPTION
**Pull request by zhaozg**

### CODE REVIEW

**Comment by squeek502** [comment]
CI failures are due to https://github.com/luvit/lit/issues/253

**Comment by zhaozg** [comment]
On MacOS `https://github.com/luvit/luvi/releases/download/v2.9.1/luvi-regular-Darwin_x86_64`
return 404

on Windows error with C:/Users/tim/luvi/src/lua/luvibundle.lua:321: bundle:deps/require.lua: cannot load incompatible bytecode

Please check ci output

**Comment by squeek502** [comment]
If you revert the Lit version back to v3.5.4 it should work now (https://github.com/luvit/lit/issues/253 has been fixed).

We should probably wait until luvit/luvi#200 is fixed before bumping the Lit/Luvi version in Luvit.

File: deps/process.lua [hunk 1]
```diff
@@ -56,18 +56,19 @@ local lenv = {}
...
+    index = index + 1
+    local name = keys[index]
+    if name then
+      return name, env.get(name)
     end
-  end,
+  end, keys, nil
+end
+
+setmetatable(lenv, {
```
Comment (squeek502) [inline deps/process.lua]: Should still set `__pairs` to `lenv.iterate` for backwards compatibility

**Comment by zhaozg** [comment]
Self test pass, ready to merge @squeek502 

Test platform
```
 ✘ zhaozg@zhaozgdeMacBook-Pro  ~/work/luvit/luvit   master  luvi -v
luvi v2.9.2-33-g6636ca4
rex: 8.43 2019-02-23
libuv: 1.27.0
ssl: OpenSSL 1.0.2s-dev  xx XXX xxxx, lua-openssl 0.7.5
 ✘ zhaozg@zhaozgdeMacBook-Pro  ~/work/luvit/luvit   master  uname -a
Darwin zhaozgdeMacBook-Pro.local 18.2.0 Darwin Kernel Version 18.2.0: Thu Dec 20 20:46:53 PST 2018; root:xnu-4903.241.1~1/RELEASE_X86_64 x86_64
```

File: deps/process.lua (hunk 1 above)
Comment (zhaozg) [reply deps/process.lua]: Maybe luajit compile without COMPAT to lua 5.2, so change this, and now make __pairs to lenv.iterate.

**Comment by rphillips** [comment]
nice. thank you!

**Pull request merged**